Gems/ROS2/Code/Source/ImGui/SensorImGui.cpp

```diff
@@ -0,0 +1,144 @@
+            for (const auto& [key, value] : sensorConfiguration.m_publishersConfigurations)
+            {
+                label += "Topic: " + value.m_topic + "\n";
+            }
+
+            AZStd::string buttonText = sensorConfiguration.m_publishingEnabled ? "Disable Publishing" : "Enable Publishing";
+
+            auto buttonCallback = [sharedSensorComponentTickBased, sharedSensorComponentPhysicsBased, sensorConfiguration]() mutable
+            {
+                sensorConfiguration.m_publishingEnabled = !sensorConfiguration.m_publishingEnabled;
+                if (sharedSensorComponentTickBased)
+                {
+                    sharedSensorComponentTickBased->SetSensorConfiguration(sensorConfiguration);
+                }
+                else if (sharedSensorComponentPhysicsBased)
+                {
+                    sharedSensorComponentPhysicsBased->SetSensorConfiguration(sensorConfiguration);
+                }
+            };
+
+            m_imGuiElements.emplace_back(label, buttonText, true /* hasButton */, false /* hasCheckbox */, buttonCallback);
+        }
+        return m_imGuiElements;
+    }
+
+    void SensorImGui::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<SensorImGui, AZ::Component>()->Version(0);
+            if (auto* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<SensorImGui>("ROS2 Sensor ImGui", "ImGui for ROS 2 Sensors")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"));
+            }
+        }
+    }
+
+    SensorImGui::SensorImGui()
+    {
+    }
+
+    SensorImGui::~SensorImGui()
+    {
+    }
+
+    void SensorImGui::Activate()
+    {
+        ImGui::ImGuiUpdateListenerBus::Handler::BusConnect();
+        ROS2::ImGuiRequestBus::Handler::BusConnect(GetEntityId());
+        entity = GetEntity();
+        m_imGuiElements.clear();
+    }
+
+    void SensorImGui::Deactivate()
+    {
+        ImGui::ImGuiUpdateListenerBus::Handler::BusDisconnect();
+        ROS2::ImGuiRequestBus::Handler::BusDisconnect();
+    }
+
+    void SensorImGui::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("ROS2SensorImGuiComponent"));
+    }
+
+    void SensorImGui::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("ROS2SensorImGuiComponent"));
+    }
+
+    void SensorImGui::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+    }
+
+    void SensorImGui::GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent)
```

Comment:
Do not implement if there is no optional services (are you sure?).

Reply:
.